### PR TITLE
remove dead code in rbac helper

### PR DESCRIPTION
**Pull request by xilabao**
**What this PR does / why we need it**:
validated in `case len(r.PolicyRule.NonResourceURLs) > 0:` .

**Which issue this PR fixes**: #49540

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/rbac/helpers.go [hunk 1]
```diff
@@ -215,9 +215,8 @@ func (r *PolicyRuleBuilder) Rule() (PolicyRule, error) {
 			return PolicyRule{}, fmt.Errorf("non-resource rule may not have apiGroups, resources, or resourceNames: %#v", r.PolicyRule)
 		}
 	case len(r.PolicyRule.Resources) > 0:
-		if len(r.PolicyRule.NonResourceURLs) != 0 {
-			return PolicyRule{}, fmt.Errorf("resource rule may not have nonResourceURLs: %#v", r.PolicyRule)
-		}
+		// resource rule may not have nonResourceURLs
```
Comment (supereagle) [inline pkg/apis/rbac/helpers.go]: Is this comment not needed? If comes here, `len(r.PolicyRule.NonResourceURLs)` is 0.

**Comment by brendandburns** [comment]
/lgtm

/release-note-none

LGTM + release-note-none from the bulk LGTM tool

**Comment by cblecker** [comment]
```
I0622 11:13:52.505]   /go/src/k8s.io/kubernetes/staging/src/k8s.io/client-go is out of date. Please run hack/update-staging-client-go.sh
```
/ok-to-test

**Pull request merged**